lib/puppet_x/puppetlabs/dsc_type_helpers.rb

```diff
@@ -126,6 +126,24 @@ def self.validate_mof_type(mof_type, embeddedinstance_name, name, value)
           end
         end
       end
+
+      def self.reboot_relationship(resource)
+        reboot_resource = resource.catalog.resource(:reboot, 'dsc_reboot')
+        return [] if !reboot_resource
+
+        # if an edge already exists from Reboot[dsc_reboot] to this resource, we have problems
```

Comment:
I opted to rely on Puppets cyclic dependency checking which is certainly more robust than checking for a single edge, which is what we were doing.

This simplifies the edge building code a little bit as a side effect.